osbuild.py

```diff
@@ -142,20 +140,19 @@ def run_stage(self, stage, tree, interactive=False):
             "output": r.stdout
         }
 
-    def run_assembler(self, assembler, tree, output_dir=None, interactive=False):
+    def run_assembler(self, name, options, resources, tree, output_dir=None, interactive=False):
```

Comment:
Since we have `Assembler.run` now, could we just move the contents of this function there? (Same for `Stage.run`)

Reply:
Absolutely. I'll rebase to fix the other comments, and add a patch on top to do that.